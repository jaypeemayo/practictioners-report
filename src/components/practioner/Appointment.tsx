import * as React from "react";
import AppointmentDetails from "./AppointmentDetails";

export interface IAppointmentProps {
    date: string,
    cost: string,
    revenue: string
}

export interface IAppointmentState {
    isAppointmentDetailsShown: boolean;
}

export default class Appointment extends React.Component<IAppointmentProps, IAppointmentState> {
    state: IAppointmentState = { isAppointmentDetailsShown: false };
    onClick = () => {
        this.setState({isAppointmentDetailsShown: !this.state.isAppointmentDetailsShown})
    }
    render() {
        return (
            <>
                <tr onClick={this.onClick}>
                    <th scope="row">{this.props.date}</th>
                    <td>{this.props.cost}</td>
                    <td>{this.props.revenue}</td>

                </tr>
                {this.state.isAppointmentDetailsShown && <tr>
                    <td colSpan={3}>
                        <AppointmentDetails />
                    </td>
                </tr>}
            </>
        );
    }
}
