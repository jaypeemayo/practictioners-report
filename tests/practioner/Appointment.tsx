import * as React from 'react';
import * renderer from 'react-test-renderer';
import Appointment from '../../src/components/practioner/Appointment';

it('should render appointment', () => {

    let appointment = {
        clientName: "string",
        cost: 100,
        date: new Date(2000),
        duration: 200,
        id: 1,
        revenue:300,
        type: "Type A"
    };

    const component = renderer.create(
        <Appointment {...appointment}/>,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});


it('should toggle when clicking appointment', () => {

    let appointment = {
        clientName: "string",
        cost: 100,
        date: new Date(2000),
        duration: 200,
        id: 1,
        revenue:300,
        type: "Type A"
    };

    const component = renderer.create(
        <Appointment {...appointment}/>,
      );

      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      tree.props.onClick();
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});

