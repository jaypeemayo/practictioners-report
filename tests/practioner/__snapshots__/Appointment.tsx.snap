// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render appointment 1`] = `
<tr
  onClick={[Function]}
>
  <th
    scope="row"
  >
    01/01/1970
  </th>
  <td>
    10:00 AM
  </td>
  <td>
    100
  </td>
  <td>
    300
  </td>
  <td>
    <svg
      attr={
        Object {
          "viewBox": "0 0 448 512",
        }
      }
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
      />
    </svg>
  </td>
</tr>
`;

exports[`should toggle when clicking appointment 1`] = `
<tr
  onClick={[Function]}
>
  <th
    scope="row"
  >
    01/01/1970
  </th>
  <td>
    10:00 AM
  </td>
  <td>
    100
  </td>
  <td>
    300
  </td>
  <td>
    <svg
      attr={
        Object {
          "viewBox": "0 0 448 512",
        }
      }
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
      />
    </svg>
  </td>
</tr>
`;

exports[`should toggle when clicking appointment 2`] = `
Array [
  <tr
    onClick={[Function]}
  >
    <th
      scope="row"
    >
      01/01/1970
    </th>
    <td>
      10:00 AM
    </td>
    <td>
      100
    </td>
    <td>
      300
    </td>
    <td>
      <svg
        attr={
          Object {
            "viewBox": "0 0 448 512",
          }
        }
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
      >
        <path
          d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
        />
      </svg>
    </td>
  </tr>,
  <tr>
    <td
      colSpan={5}
    >
      <div
        className="border border-secondary"
      >
        <h6
          className="p-3 mb-2 bg-secondary text-white"
        >
          Appointments Detail
        </h6>
        <div
          className="row"
        >
          <div
            className="col-6 text-right display"
          >
            <strong>
              Client Name:
            </strong>
          </div>
          <div
            className="col-6"
          >
            string
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-6 text-right"
          >
            <strong>
              Appointment Type:
            </strong>
          </div>
          <div
            className="col-6"
          >
            Type A
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-6 text-right"
          >
            <strong>
              Duration:
            </strong>
          </div>
          <div
            className="col-6"
          >
            200
             min
          </div>
        </div>
      </div>
    </td>
  </tr>,
]
`;
